packages/cosmwasm-stargate/src/signingcosmwasmclient.spec.ts

```diff
@@ -1191,6 +1191,73 @@ describe("SigningCosmWasmClient", () => {
 
         client.disconnect();
       });
+
+      it("works with a custom timeoutHeight", async () => {
+        pendingWithoutWasmd();
+        const wallet = await DirectSecp256k1HdWallet.fromMnemonic(alice.mnemonic, { prefix: wasmd.prefix });
+        const client = await SigningCosmWasmClient.connectWithSigner(wasmd.endpoint, wallet, {
+          ...defaultSigningClientOptions,
+        });
+
+        const msg = MsgSend.fromPartial({
+          fromAddress: alice.address0,
+          toAddress: alice.address0,
+          amount: [coin(1, "ucosm")],
+        });
+        const msgAny: MsgSendEncodeObject = {
+          typeUrl: "/cosmos.bank.v1beta1.MsgSend",
+          value: msg,
+        };
+        const fee = {
+          amount: coins(2000, "ucosm"),
+          gas: "222000", // 222k
+        };
+        const memo = "Use your power wisely";
+        const height = await client.getHeight();
+        const signed = await client.sign(alice.address0, [msgAny], fee, memo, undefined, BigInt(height + 3));
+
+        // ensure signature is valid
+        const result = await client.broadcastTx(Uint8Array.from(TxRaw.encode(signed).finish()));
+        assertIsDeliverTxSuccess(result);
+
+        client.disconnect();
+      });
+
+      it("fails with past timeoutHeight", async () => {
+        pendingWithoutWasmd();
+        const wallet = await DirectSecp256k1HdWallet.fromMnemonic(alice.mnemonic, { prefix: wasmd.prefix });
+        const client = await SigningCosmWasmClient.connectWithSigner(wasmd.endpoint, wallet, {
+          ...defaultSigningClientOptions,
+        });
+
+        const msg = MsgSend.fromPartial({
+          fromAddress: alice.address0,
+          toAddress: alice.address0,
+          amount: [coin(1, "ucosm")],
+        });
+        const msgAny: MsgSendEncodeObject = {
+          typeUrl: "/cosmos.bank.v1beta1.MsgSend",
+          value: msg,
+        };
+        const fee = {
+          amount: coins(2000, "ucosm"),
+          gas: "222000", // 222k
+        };
+        const memo = "Use your power wisely";
+        const height = await client.getHeight();
+        const signed = await client.sign(alice.address0, [msgAny], fee, memo, undefined, BigInt(height - 1));
+
+        try {
+          await client.broadcastTx(Uint8Array.from(TxRaw.encode(signed).finish()));
+        } catch (e: any) {
+          assert(e.code === 30);
+          return;
```

Comment:
Doesn't this return the function without disconnecting?

You can also write expected exceptions like this. Also check out other `expectAsync` usages in the codebase

```
      await expectAsync(
        client.queryContractSmart(nonExistentAddress, { verifier: {} }),
      ).toBeRejectedWithError(/no such contract/i);
```

Reply:
It should run finally block, but anyway `expectAsync` is the function I was looking for. Thanks :) 

btw I like when things go full circle https://github.com/jasmine/jasmine/issues/1625#issuecomment-444810001

Reply:
Thanks, looks nicer!